signalfx-tracing/signalfx-python-tracing/django/client.py

```diff
@@ -0,0 +1,111 @@
+# IMPORTANT additions to a client file:
+# from signalfx_tracing import auto_instrument, create_tracer, trace
+# from signalfx_tracing.libraries import requests_config
+#
+# requests_config.propagate = True
+# tracer = create_tracer(service_name='<APP_NAME>')
+# auto_instrument(tracer)
+#
+# #Then in your main function, at the end of all function calls to be traced:
+# tracer.close()
+#
+# These are noted below with additional explanations.
+
+
+import logging
+import os
+import requests
+from requests.auth import HTTPBasicAuth
+from signalfx_tracing import auto_instrument, create_tracer, trace
+# Important: Needed to propagate traces
+from signalfx_tracing.libraries import requests_config
+
+
+# Create and configure logger
+if not os.path.exists('log'):
+    os.makedirs('log')
+logging.basicConfig(level='DEBUG',
+                    filename="log/client_log.log",
+                    format='%(asctime)s %(message)s',)
+
+
+# create an instance of the SignalFx tracer
+# and configure auto-instrumentation of the tracing instance
+tracer = create_tracer(service_name='BlogClient')
+auto_instrument(tracer)
+
+
+# Setting requests_config.propagate to 'True'  ensures
+# that the relational information between traces is maintained
+# and forwarded.
+requests_config.propagate = True
+
+
+class BlogClient:
+    def __init__(self, url, username, password):
+        self.url = url
+        self.session = requests.Session()
+        self.session.auth = HTTPBasicAuth(username, password)
+
+    @trace
+    def create_blog(self, new_title, new_description, new_body):
+        response = requests.post(self.url,
+                                 json=dict(title=new_title,
+                                           description=new_description,
+                                           body=new_body),
+                                 auth=self.session.auth, )
+
+        logging.info('INFO - Created status: {}'.format(response.status_code))
+        return response.json().get('id')
+
+    @trace
+    def list_blogs(self):
+        response = requests.get(self.url)
+        logging.info('INFO - Viewed status: {}'.format(response.status_code))
+        return response.json()
+
+    @trace
+    def get_blog(self, blog_id):
+        url_to_read = self.url.format(blog_id)
+        response = requests.get(url_to_read)
+        logging.info('INFO - Read status: {}'.format(response.status_code))
+        return response.json()
+
+    @trace
+    def update_blog(self, blog_id, new_title):
+        url_to_update = self.url + '{}/'.format(blog_id)
+        response = requests.patch(url_to_update,
+                                  json=dict(title=new_title),
+                                  auth=self.session.auth, )
+        logging.info('INFO - Updated status: {}'.format(response.status_code))
+        return response.json()
+
+    @trace
+    def delete_blog(self, blog_id):
```

Comment:
mind checking the status code and returning a boolean?

Reply:
Sure, can do that

Reply:
Done